core/tracker/tracker.go

```diff
@@ -132,17 +132,62 @@ func analyseDutyFailed(duty core.Duty, es []event) (bool, component, string) {
 	})
 
 	if len(events) == 0 {
-		return false, sentinel, "No events to analyse"
+		return false, sentinel
 	}
 
-	if events[0].component == sigAgg {
-		// Duty completed successfully
+	c := events[0].component
+
+	return c != sigAgg, c + 1
```

Comment:
this doesn't match the comment ` If the duty didn't get stuck, it returns the sigAgg component`